tf2_ros_py/tf2_ros/buffer.py

```diff
@@ -48,41 +60,49 @@ class Buffer(tf2.BufferCore, tf2_ros.BufferInterface):
 
     Stores known frames and offers a ROS service, "tf_frames", which responds to client requests
     with a response containing a :class:`tf2_msgs.FrameGraph` representing the relationship of
-    known frames. 
+    known frames.
     """
 
-    def __init__(self, cache_time = None, node = None):
+    def __init__(
+        self,
+        cache_time: Optional[Duration] = None,
+        node: Optional[Node] = None
+    ) -> None:
         """
         Constructor.
 
-        :param cache_time: (Optional) Duration object describing how long to retain past information in BufferCore.
-        :param node: (Optional) If node create a tf2_frames service, It responses all frames as a yaml
+        :param cache_time: Duration object describing how long to retain past information in BufferCore.
+        :param node: If node create a tf2_frames service, It responses all frames as a yaml
         """
         if cache_time is not None:
             tf2.BufferCore.__init__(self, cache_time)
         else:
             tf2.BufferCore.__init__(self)
         tf2_ros.BufferInterface.__init__(self)
 
-        self._new_data_callbacks = []
-        self._callbacks_to_remove = []
+        self._new_data_callbacks: List[Callable[[None], None]] = []
+        self._callbacks_to_remove: List[Callable[[None], None]] = []
         self._callbacks_lock = threading.RLock()
 
         if node is not None:
             self.srv = node.create_service(FrameGraph, 'tf2_frames', self.__get_frames)
 
-    def __get_frames(self, req, res):
-       return FrameGraph.Response(frame_yaml=self.all_frames_as_yaml())
+    def __get_frames(
+        self,
+        req: FrameGraphSrvRequest,
+        res: FrameGraphSrvResponse
+    ) -> FrameGraphSrvResponse:
+        return FrameGraph.Response(frame_yaml=self.all_frames_as_yaml())
 
-    def set_transform(self, *args, **kwargs):
+    def set_transform(self, *args: Tuple[TransformStamped, str], **kwargs: Any) -> None:
```

Comment:
@clalancette  Should the signature of `set_transform()` and `set_transform_static()` be respectfully:

```python
def set_transform(self, *args):
    ...

def set_transform_static(self, *args):
    ...
```

I ask as I see that the function signature at https://github.com/ros2/geometry2/blob/a4f152f67d09b50d6d6450bb00c44512e52a880c/tf2_py/src/tf2_py.cpp#L656 and also see that `PyArg_ParseTuple()` is used as opposed to `PyArg_ParseTupleAndKeywords()`

Reply:
Good point.  But this is honestly more than I know about Python; maybe @sloretz could chime in here and give an opinion.

The context is that we are trying to add type annotations to the tf2_ros_py code, and we're seeing that while `tf2_ros_py::Buffer::set_transform` takes a `*args` and `**kwargs`, it is just a thin wrapper around `tf2_py::BufferCore::set_transform`.  The latter *seems* to only take a `*args`, though I'm not sure about that.  Any thoughts here?

Reply:
Please note that I used `Any` type for `**kwargs` as a placeholder for the moment. Will change accordingly based on @sloretz response.

Reply:
Since we have a separate issue for this now (#316), I'm going to mark this conversation resolved.  We'll follow-up there.